Add details about `table-column-count-limit` and `index-limit` (#4949)
pingcap/docs: tidb-limitations.md

## Limitations on a single table

| Type       | Upper limit (default value)  |
|:----------|:----------|

> | Columns   | The default is 1017, and the upper limit can be adjusted to 4096.      |

Reviewer: ```suggestion
| Columns   | Defaults to 1071 and can be adjusted up to 4096     |
```

> | Indexes   | The default is 64, and the upper limit can be adjusted to 512.        |

Reviewer: ```suggestion
| Indexes   |  Defaults to 64 and can be adjusted up to 512        |
```

> | Indexes   | The default is 64, and the upper limit can be adjusted to 512.        |

Reviewer: ```suggestion
| Indexes   |  Defaults to 64 and can be adjusted up to 512        |
```

| Rows      | unlimited |
| Size      | unlimited |
| Partitions | 1024      |

> * The upper limit of Columns can be modified by [`table-column-count-limit`](/tidb-configuration-file.md#table-column-count-limit-new-in-v500-rc).

Reviewer: ```suggestion
* The upper limit of `Columns` can be modified via [`table-column-count-limit`](/tidb-configuration-file.md#table-column-count-limit-new-in-v500-rc).
```

> * The upper limit of Indexes can be modified by [`index-limit`](/tidb-configuration-file.md#index-limit-new-in-v500-rc).

Reviewer: ```suggestion
* The upper limit of `Indexes` can be modified via [`index-limit`](/tidb-configuration-file.md#index-limit-new-in-v500-rc).
```
